subsys/dfu/boot/mcuboot.c

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2016-2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_DOMAIN "fota/mcuboot"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_FOTA_LEVEL
+#include <logging/sys_log.h>
+
+#include <stddef.h>
+#include <errno.h>
+#include <string.h>
+#include <flash.h>
+#include <zephyr.h>
+#include <init.h>
+
+#include "boot/mcuboot.h"
+
+/*
+ * Helpers for image trailer, as defined by mcuboot.
+ * Image trailer consists of sequence of fields:
+ *   u8_t copy_done
+ *   u8_t pading_1[BOOT_MAX_ALIGN - 1]
+ *   u8_t image_ok
+ *   u8_t pading_2[BOOT_MAX_ALIGN - 1]
+ *   u8_t magic[16]
+ */
+
+/* Strict defines: Defines in block below must be equal to coresponding
+ * mcuboot defines */
+#define BOOT_MAX_ALIGN 8
+#define BOOT_MAGIC_SZ  16
+#define BOOT_FLAG_SET 0x01
+#define BOOT_FLAG_UNSET 0xff
+/* end_of Strict defines */
+
+#define BOOT_MAGIC_GOOD  1
+#define BOOT_MAGIC_BAD   2
+#define BOOT_MAGIC_UNSET 3
+
+#define BOOT_FLAG_IMAGE_OK 0
+#define BOOT_FLAG_COPY_DONE 1
+
+#define COPY_DONE_OFFS(bank_offs) (bank_offs + FLASH_BANK_SIZE -\
```

Comment:
You might check the generated code, but at least in the past __packed had a significant impact on the generated code. It might not matter in this case, since you're only accessing a byte field, but in the general case, it does revert to reading/writing all fields with byte accesses.